Add example for proxying upstream with custom routes (#14374)
vercel/next.js: examples/custom-routes-proxying/README.md

# Custom Routes Proxying Example

> This example shows the most basic example using Next.js' new custom routes feature to proxy requests to an upstream server. We have 3 pages: `pages/index.js`, `pages/about.js`, and `pages/hello/[slug].js`. All of these pages will be matched against before requests are proxied to the upstream server.

Reviewer: ```suggestion
This example shows the most basic example using Next.js' new custom routes feature to proxy requests to an upstream server. We have 3 pages: `pages/index.js`, `pages/about.js`, and `pages/hello/[slug].js`. All of these pages will be matched against Next.js and any other path will be proxied to the upstream server.
```
